src/pytest_mypy.py

```diff
@@ -89,7 +89,10 @@ class MypyItem(pytest.Item, pytest.File):
     MARKER = 'mypy'
 
     def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
+        super(MypyItem, self).__init__(*args, **kwargs)
+        if hasattr(self, "_nodeid"):
```

Comment:
I'm not too keen on using a private attribute for this, but there seems to be another way:
``` python
MypyItem(path, parent, nodeid=str(path) + '::MYPY')
```
There are still a couple issues with that, though:
- It uses the full path instead of the normal relative path.
- It isn't available in all versions of Pytest the this project supports.

I'm not exactly sure what to do about this yet 🤔
Worst case, we can drop support for `pytest < 3` in 1.0.0, but the current plan is to wait for Mypy 1.0.0 before releasing that (which might be awhile).